lib/ui/home/home_page.dart

```diff
@@ -43,37 +49,44 @@ class HomePageState extends State<HomePage> {
         textAlign: TextAlign.center,
       );
 
-  Widget _buildExpandedContent() => Expanded(
+  Widget _buildExpandedContent(BuildContext context) => Expanded(
         child: Column(
           children: <Widget>[
             Expanded(child: Container()),
             GestureDetector(
-              onTap: () {
-                setState(() {});
-              },
-              child: FutureBuilder<PollutionStation>(
-                  future: _pollutionRepository.getFirstStation(),
-                  builder: (context, snapshot) {
-                    if (snapshot.connectionState == ConnectionState.waiting) {
-                      return CircularProgressIndicator();
-                    } else if (snapshot.hasError) {
-                      return Text(ApiExceptionMapper.toErrorMessage(snapshot.error));
-                    } else {
-                      final station = snapshot.data;
-                      return Text('${station.stationName}');
-                    }
-                  }),
-            ),
+                onTap: () {
+                  setState(() {});
+                },
+                child: _buildStationName()),
             AnimatedPercentageWidget(
               fromValue: 0,
               toValue: 310,
             ),
-            Text(Strings.airQualityNorm, style: Theme.of(context).textTheme.subtitle),
+            Text(Strings.airQualityNorm,
+                style: Theme.of(context).textTheme.subtitle),
             _buildDetailsWidget(),
           ],
         ),
       );
 
+  Widget _buildStationName() {
+    ProviderModelAsyncResult<PollutionStation, Exception> firstStation = context
+        .select((PollutionProviderModel model) => model.value.firstStation);
+
+    if (firstStation.hasData) {
+      if (!firstStation.isLoading) {
+        if (firstStation.result.isValue) {
+          return Text('${firstStation.result.asValue.value.stationName}');
+        } else if (firstStation.result.isError) {
+          return Text(ApiExceptionMapper.toErrorMessage(
+              firstStation.result.asError.error));
+        }
+      }
+    }
```

Comment:
```suggestion
    if (firstStation.hasData && !firstStation.isLoading) {
      if (firstStation.result.isValue) {
        return Text('${firstStation.result.asValue.value.stationName}');
      } else if (firstStation.result.isError) {
        return Text(ApiExceptionMapper.toErrorMessage(
            firstStation.result.asError.error));
      }
    }
```
hmm? mniej zagnieżdżeń

Reply:
Ogólnie zmieniłbym to na `failFast` zamiast zagnieżdżeń.

Reply:
@MiSikora  Pod sam koniec jesli .result nie bedzie errorem ani value bede musial cos zwrocic (wedlug analyzera), czy ten case jakos chcemy zaznaczyc ze tam nigdy nie trafimy?
`if (!firstStation.hasData || firstStation.isLoading) {
      return CircularProgressIndicator();
    }

    if (firstStation.result.isValue) {
      return Text('${firstStation.result.asValue.value.stationName}');
    } else if (firstStation.result.isError) {
      return Text(
          ApiExceptionMapper.toErrorMessage(firstStation.result.asError.error));
    } else {
      return CircularProgressIndicator();
    }`

Reply:
`if (!firstStation.hasData || firstStation.isLoading) {
      return CircularProgressIndicator();
    }

    if (firstStation.result.isValue) {
      return Text('${firstStation.result.asValue.value.stationName}');
    } else if (firstStation.result.isError) {
      return Text(
          ApiExceptionMapper.toErrorMessage(firstStation.result.asError.error));
    } else {
      return CircularProgressIndicator();
    }`